crates/iota-indexer/src/indexer_reader.rs

```diff
@@ -1579,22 +1579,34 @@ impl IndexerReader {
     }
 
     pub fn get_latest_move_call_metrics(&self) -> IndexerResult<MoveCallMetrics> {
+        let latest_epoch = run_query!(&self.pool, |conn| {
+            move_call_metrics::table
+                .select(move_call_metrics::dsl::epoch)
+                .order(move_call_metrics::dsl::epoch.desc())
+                .limit(1)
+                .first::<i64>(conn)
+        })
+        .map_err(|_| IndexerError::PostgresRead("record not found".to_string()))?;
+
         let latest_3d_move_call_metrics = run_query!(&self.pool, |conn| {
             move_call_metrics::table
+                .filter(move_call_metrics::dsl::epoch.eq(latest_epoch))
```

Comment:
Using a subquery here would avoid the need for an additional query in L1582 to fetch the latest epoch. Something equivalent to `WHERE epoch = (SELECT MAX(epoch) from move_call_metrics WHERE day = 3)`

Note that we should also select the `MAX(epoch)` for the desired `day` (in this case 3).

Reply:
Fixed by: https://github.com/iotaledger/iota/pull/5648

Running some more tests and then we should be good.